agent-inject/agent/annotations_test.go

```diff
@@ -473,6 +473,66 @@ func TestTemplateShortcuts(t *testing.T) {
 	}
 }
 
+func TestSecretTemplateSourceAnnotations(t *testing.T) {
+	tests := []struct {
+		annotations            map[string]string
+		expectedKey            string
+		expectedTemplate       string
+		expectedTemplateSource string
+	}{
+		{
+			map[string]string{
```

Comment:
I think test coverage on a mixed case would be good to have. For example, using a template file for a large template but using a custom template via annotation for another.

Reply:
I have added an extra test for this mixed case.

Reply:
Thanks!